### DISCUSSION TITLE
Maintenance log

### DISCUSSION THREAD

**Discussion by cwolock** [statdivlab/rigr#43] [General]
This Discussion reflects the current status of all major functions in `rigr`. Specific bugs/enhancements live in Issues; this post is meant to be a general tracking tool.

**Comment by cwolock** [comment]
`descrip`

**Reply by taylorokonek** [reply]
Basic unit tests written and additional error handling added. Minor changes to output for non-Surv variables and removal of version parameter. Documentation updated as well. All helper functions located in descrip_utils.R, and everything is documented well. Overall, this function is good to go barring any bugs we find in the future or additional enhancements we want.

**Comment by cwolock** [comment]
`regress`

**Reply by taylorokonek** [reply]
All helper functions for regress moved into regress_utils.R. Unit tests in progress. A few bugs and enhancements remain (#39 #12 #11). Following discussion #36, we will be removing gee and survival capabilities from the function to make it more maintainable moving forward (in progress). Further documentation needs to be done in regress_utils.R (#38), and many of the helper functions could likely be replaced with `stringr` functions.

**Reply by taylorokonek** [reply]
Many more unit tests written (coverage is about 75% at the moment). The main `regress` function is documented line by line in the code now. All of the survival capabilities should be removed at this point, some gee code chunks still remain but will be deleted soon. A lot of bugs and coefficient labeling in output has been fixed, just a few use cases left to address (see #68, for example). Have yet to look into `lspline`, `dummy`, and `polynomial` use cases in `U`.

**Reply by taylorokonek** [reply]
`lspline` functionality has been removed, `dummy` functionality has been debugged, and more unit tests have been written. All gee capabilities should now be removed. Most bugs have been addressed. Looking into `polynomial` usage, enhancements, and further unit testing remains.

**Reply by taylorokonek** [reply]
`polynomial` functionality is working, intercepts can now be removed from models with a "-1" in the formula _or_ specifying _intercept = FALSE_, and multi-way (three-way or more) interactions are now functional

**Comment by cwolock** [comment]
`wilcoxon`

**Reply by cwolock** [reply]
Basic unit tests written and major (known) bugs fixed. Could use additional unit tests for output not provided by `stats` package's `wilcox.test` function, but as of now the comparable pieces match `wilcox.test`.

**Comment by cwolock** [comment]
`ttest`

**Reply by cwolock** [reply]
Unit tests for error handling written. Substantial bugs remain (see #42 #41 #40), although so far they do not affect inference.

**Reply by cwolock** [reply]
Unit tests for return values written. This includes one-sample, two-sample matched, two-sample unmatched, assuming equal and unequal variances, etc. Also streamlined the code a bit - eliminated the `strat` parameter which was identical to `by`. If we decide to split proportion tests off from `ttest`, then unit tests will be essentially done for this function.

**Comment by cwolock** [comment]
`ttesti`

**Reply by cwolock** [reply]
I reworked the code for this so that `ttesti` returns an object of class `ttesti` which then has a `print` method. I also harmonized the printed output with `ttest`.

**Reply by cwolock** [reply]
Unit tests for error handling and correct output are finished.

**Comment by taylorokonek** [comment]
`bplot`

**Reply by taylorokonek** [reply]
Currently, work on `bplot` has been put on hold since other functions (such as `regress`, `ttest`, etc.) are higher priority. We've discussed converting `bplot` so that it produces ggplot objects rather than base R plots, and I have a drafted version of the function on my personal machine. The function will have the same capabilities as the current `bplot` function, and users will not need to know ggplot in order to use it (it will remain a wrapper). We've also discussed changing some of the default parameters so that the plots are less crowded by default.

**Comment by cwolock** [comment]
`lincom`

**Reply by cwolock** [reply]
I have made some minor changes to `lincom`; in general, the function seems to be in good working order. I rewrote some of the code so that `lincom` returns an object versus just printing output. I also removed the `overall` test (although I can easily put it back in - see #62). Printing capabilities have been moved to `lincom_utils` to be consistent with other functions. Next task is unit testing.

**Reply by cwolock** [reply]
Unit tests for error handling and basic functionality have been written. I fixed a few minor bugs in the printed output.

**Comment by cwolock** [comment]
`proptest`

**Reply by cwolock** [reply]
This is a new function building off of the ability of old `ttest` to do tests of proportions. At the moment, `proptest` can do a one-sample proportion test (either approximate or exact, based on the binomial probabilities) or a two-sample test comparing two proportions (approximate only). Unit tests to be added.

**Reply by cwolock** [reply]
Unit tests have been added and code coverage is 100%. There are some enhancements that could be made - see #58.

**Comment by cwolock** [comment]
`proptesti`

**Reply by cwolock** [reply]
Same functionality as `proptest`, with with counts of successes and trials as opposed to binary data. Unit tests to be added.

**Reply by cwolock** [reply]
Unit tests have been added and code coverage is 100%. There are some enhancements that could be made - see #58.
